Userland/Libraries/LibWeb/HTML/Navigable.h

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2022, Andreas Kling <kling@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/DeprecatedString.h>
+#include <LibJS/Heap/Cell.h>
+#include <LibWeb/Forward.h>
+
+namespace Web::HTML {
+
+// https://html.spec.whatwg.org/multipage/document-sequences.html#navigable
+class Navigable : public JS::Cell {
+    JS_CELL(Navigable, JS::Cell);
+
+public:
+    virtual ~Navigable() override;
+
+    DeprecatedString id() const;
```

Comment:
Let's not introduce more uses of `DeprecatedString` in new code. Also should return a const ref.